src/main.ts

```diff
@@ -182,7 +160,18 @@ async function asyncForEach<T>(
 
 async function run(): Promise<void> {
   const argocd = await setupArgoCDCommand();
-  const apps = await getApps();
+  
+  let apps = await getApps();
+  // Filter to only Apps sourced from the repo that the action is running for.
+  // It will have access to the files for these to run `diff --local` on.
+  apps = filterAppsByRepo(apps, `${github.context.repo.owner}/${github.context.repo.repo}`)
+  // Filter Apps to those following the repo trunk, since that is what the PR is
+  // comparing against (in most cases).
+  apps = filterAppsByTargetRevision(apps)
+  // Filter Apps that are in an excluded path, if any are provided.
+  if (EXCLUDE_PATHS != []) {
+    apps = filterAppsByExcludedPath(apps, EXCLUDE_PATHS)
+  }
```

Comment:
```suggestion
  let apps = await getApps();

  // We can only run `diff --local` on files that are for this current repo.
  apps = filterAppsByRepo(apps, `${github.context.repo.owner}/${github.context.repo.repo}`);
  
  // Filter Apps to those following the repo trunk, since that is what the PR is
  // comparing against (in most cases).
  apps = filterAppsByTargetRevision(apps);
  
  apps = filterAppsByExcludedPath(apps, EXCLUDE_PATHS);
```

Better spacing and revised the comments to just explain the why and not the what/how.

In case of excluded path, it's much more readable if you move the IF statement over to the filter function as it'll simply return the same object if the exclude paths is empty